rskj-core/src/test/java/co/rsk/peg/feeperkb/constants/FeePerKbConstantsTest.java

```diff
@@ -52,7 +54,7 @@ private static Stream<Arguments> getMaxFeePerKbProvider() {
     void getFeePerKbChangeAuthorizer(FeePerKbConstants feePerKbConstants,
         AddressBasedAuthorizer expectedFeePerKbChangeAuthorizer) {
         AddressBasedAuthorizer actualFeePerKbChangeAuthorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
-        assertEquals(expectedFeePerKbChangeAuthorizer, actualFeePerKbChangeAuthorizer);
+        assertThat(expectedFeePerKbChangeAuthorizer, samePropertyValuesAs(actualFeePerKbChangeAuthorizer));
```

Comment:
This is the correct way to compare the value between 2 objects with an `assert`. I think that overriding the `equal`, and `hashCode` methods isn't worth it in this case since we might face the same situation a lot of times and we already have an assert that works. 

On the other hand, a good workaround would be using AssertJ, it has a way to do it easily, but we don't have it on the project. This is an example like AssertJ implement it:

`assertThat(actualObject).usingRecursiveComparison().isEqualTo(expectedObject);`

Reply:
Interesting. I think we can also look at this as the test uncovering the lack of `equals` implementation in  AddressBasedAuthorizer class.

I'll create a separate ticket to implement it.

Thanks!